ConsistencyManagerTests/HelperClasses/ProjectionTestModel.swift

```diff
@@ -0,0 +1,120 @@
+// © 2016 LinkedIn Corp. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import Foundation
+import ConsistencyManager
+
+
+/**
+ This model is intended for unit testing the consistency manager. It simply declares some child objects and some data.
+ It is similar to TestModel but declares an extra field.
+ Therefore, it also implements mergeModel which will merge TestModel into this model.
+ */
+final class ProjectionTestModel: ConsistencyManagerModel, Equatable {
+    let id: String?
+    let data: Int?
+    let otherData: Int?
+    let children: [ProjectionTestModel]
+    let requiredModel: TestRequiredModel
+
+    init(id: String?, data: Int?, otherData: Int?, children: [ProjectionTestModel], requiredModel: TestRequiredModel) {
+        self.id = id
+        self.data = data
+        self.otherData = otherData
+        self.children = children
+        self.requiredModel = requiredModel
+    }
+
+    var modelIdentifier: String? {
+        return id
+    }
+
+    func map(transform: ConsistencyManagerModel -> ConsistencyManagerModel?) -> ConsistencyManagerModel? {
+        var newChildren: [ProjectionTestModel] = []
+        for model in children {
+            let newModel = transform(model) as? ProjectionTestModel
+            if let newModel = newModel {
```

Comment:
you can convert this line and the previous line if you'd like:

if let newModel = transform(model) as? ProjectionTestModel